pkg/commands/block_gen.go

```diff
@@ -0,0 +1,146 @@
+package commands
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+	"sort"
+	"time"
+
+	"github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"github.com/grafana/cortex-tools/pkg/bench"
+	"github.com/pkg/errors"
+	"github.com/prometheus/prometheus/pkg/labels"
+	"github.com/prometheus/prometheus/prompb"
+	"github.com/prometheus/prometheus/tsdb"
+	"gopkg.in/alecthomas/kingpin.v2"
+	"gopkg.in/yaml.v3"
+)
+
+// BlockGenCommand is the kingpin command to generate blocks of mock data.
+type BlockGenCommand struct {
+	Series     []bench.SeriesDesc `yaml:"series"`
+	Cfg        BlockGenConfig     `yaml:"block_gen"`
+	configFile string
+}
+
+type BlockGenConfig struct {
+	Interval  time.Duration `yaml:"interval"`
+	BlockSize time.Duration `yaml:"block_size"`
+	BlockDir  string        `yaml:"block_dir"`
+	MinT      int64         `yaml:"min_t"`
+	MaxT      int64         `yaml:"max_t"`
+}
+
+// Register is used to register the command to a parent command.
+func (f *BlockGenCommand) Register(app *kingpin.Application) {
+	app.Flag("config.file", "configuration file for this tool").Required().StringVar(&f.configFile)
+	app.Action(f.run)
+}
+
+func (f *BlockGenCommand) run(k *kingpin.ParseContext) error {
+	logger := log.NewLogfmtLogger(log.NewSyncWriter(os.Stderr))
+
+	content, err := os.ReadFile(f.configFile)
+	if err != nil {
+		return errors.Wrap(err, "unable to read workload YAML file from the disk")
+	}
+
+	err = yaml.Unmarshal(content, &f)
+	if err != nil {
+		return errors.Wrap(err, "unable to unmarshal workload YAML file")
+	}
+
+	if f.Cfg.BlockDir == "" {
+		var err error
+		f.Cfg.BlockDir, err = ioutil.TempDir("", "mockdata")
+		if err != nil {
+			return errors.Wrap(err, "failed to create tmp dir")
+		}
+	}
+
+	seriesSet, totalSeriesTypeMap := bench.SeriesDescToSeries(f.Series)
+	totalSeries := 0
+	for _, typeTotal := range totalSeriesTypeMap {
+		totalSeries += typeTotal
+	}
+
+	writeWorkLoad := bench.WriteWorkload{
+		TotalSeries:        totalSeries,
+		TotalSeriesTypeMap: totalSeriesTypeMap,
+		Replicas:           1,
```

Comment:
Maybe worth making this configurable? It's useful as a lever to configure the number of metrics total quickly. For example, create a workload w/ 50k metrics per replica. You can just raise the replica to have 100k, 500k, etc...

```suggestion
		Replicas:           1,
```

Reply:
Good idea, updated that: https://github.com/grafana/cortex-tools/pull/191/commits/ddeff2954cb36d628c94c6b53708fe28b7b6a8b5